### PR TITLE
core: 4.0 bug fixes

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
- Cleanup Readme for package
- Fix missing slot metadata for storybook
- Copy dist to demo apps on build
- Fix/improve link support for all button types (cds-button, cds-icon-button, cds-tag)
- Set config to prevent excessive CPU usage with vscode and yarn monorepo setup
- Fix missing focus trap on modal and check for `hidden`
- Add disabled style for tag button
- Fix close button wrapping on modal
- Fix aria hidden issue with icon button
- Update core icon to support white space offset with ratio value instead of px
- Update internal close button to better visually align within parent content box
- Fix stacking margins on lists

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "search.followSymlinks": false,
```
Comment (coryrylan) [inline .vscode/settings.json:2]: Yarn symlinks is what has caused a lot of the slowdown in vscode.

File: packages/core/src/alert/alert.element.scss [hunk 2]
```diff
@@ -138,8 +134,7 @@ cds-internal-close-button {
 
 :host([type='banner']) cds-internal-close-button {
   --color: var(--icon-color);
-  height: $cds-token-space-size-9;
-  margin-top: $cds-token-space-size-3;
+  padding-top: $cds-token-space-size-2;
```
Comment (mathisscott) [inline packages/core/src/alert/alert.element.scss:137]: Visually verified?

File: packages/core/package.js [hunk 3]
```diff
@@ -20,27 +16,91 @@ const read = dir =>
...
+}
+
+(async () => {
+  await copyAssets();
+  removeCacheFiles();
+  cleanPackageFiles();
+  updateFileVersions();
+  generateAPIMetaData();
+  await distributeBuild();
+})();
```
Comment (coryrylan) [inline packages/core/package.js:106]: Consolidated all the core packaging scripts into the single package.js. Fixes a metadata issue with Storybook not getting default slots as well as copies the dist to the demo apps for easier testing after the build has completed.

File: packages/core/src/alert/alert.element.ts [hunk 4]
```diff
@@ -259,14 +259,13 @@ export class CdsAlert extends LitElement {
...
-              ><slot name="close-button"
-                ><cds-internal-close-button
-                  icon-size="${this.type === 'banner' ? '20' : '16'}"
-                  @click="${() => this.closeAlert()}"
-                  aria-label="${this.closeIconTitle}"
-                ></cds-internal-close-button></slot
-            ></span>`
+          ? html` <slot name="close-button">
+              <cds-internal-close-button
+                icon-size="${this.type === 'banner' ? '11' : '9'}"
```
Comment (mathisscott) [inline packages/core/src/alert/alert.element.ts:264]: Why are we shrinking this icon?

File: packages/core/src/button/button.stories.ts [hunk 5]
```diff
@@ -163,31 +163,33 @@ export const links = () => {
...
+      <a href="javascript:void(0)">
+        <cds-button action="outline">this is a long link</cds-button>
+      </a>
 
-      <cds-button action="outline" size="sm">
-        <a href="javascript:void(0)">small link</a>
-      </cds-button>
+      <a href="javascript:void(0)">
+        <cds-button action="outline" size="sm">small link</cds-button>
+      </a>
```
Comment (coryrylan) [inline packages/core/src/button/button.stories.ts:192]: Based on our internal conversations this adds support to wrap buttons with an anchor. This is not a breaking change the slotted anchor behavior remains in 4.0. The base button will detect the anchor and switch to readonly mode preserving the proper semantics and a11y of the link.

The wrapping style API seems to be far more consistent/predictable with styles and easier to support framework behavior with custom routing solutions.

File: packages/core/src/internal/base/button.base.ts [hunk 6]
```diff
@@ -136,8 +142,15 @@ export class CdsBaseButton extends LitElement {
   }
 
   private updateButtonAttributes() {
-    this.isAnchor = !!this.anchor;
-    this.readonly = this.readonly || this.isAnchor;
+    this.containsAnchor = !!this.anchor;
+    this.isAnchor = this.parentElement?.tagName === 'A';
+
+    if (this.isAnchor && this.parentElement) {
```
Comment (mathisscott) [inline packages/core/src/internal/base/button.base.ts:148]: Do we have unit tests for all this?

File: packages/core/src/internal-components/close-button/close-button.element.scss [hunk 7]
```diff
@@ -17,6 +17,12 @@
   opacity: var(--opacity);
   background: var(--background);
   cursor: pointer;
+  line-height: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  min-width: $cds-token-space-size-9;
+  min-height: $cds-token-space-size-9;
```
Comment (coryrylan) [inline packages/core/src/internal-components/close-button/close-button.element.scss:25]: By providing the icon space offset ratio and the close button offset we get visual alignment without reducing the touch target of the button.

Screen Shot 2020-08-26 at 12 54 10 PM

File: packages/core/src/internal/base/button.base.ts (hunk 6 above)
Comment (coryrylan) [reply packages/core/src/internal/base/button.base.ts:146]: hmm it should, I think I goofed when I rebased onto master. I'll get them added back.

**Comment by coryrylan** [comment]
Closing and will reopen new PRs, some of my work is missing I think from a bad rebase

**Pull request closed without merging**